### DISCUSSION TITLE
Performance Gap: YOLOv11 on GTX 1650 vs. RTX 4090 – How scalable is inference?

### DISCUSSION THREAD

**Discussion by AbdulsamadHussain** [ultralytics/ultralytics#21155] [General]
Hi,

I’ve been using YOLOv11 in a PyQt6-based video processing pipeline (dashcam footage + GPS overlay + KML export). It runs, but I’m clearly hitting the ceiling on a GTX 1650 (4GB). My app performs detections and saves screenshots, outputs CSVs, and displays real-time logs.

My setup:
GPU: GTX 1650 (4GB)
CPU: Ryzen 7 5800H
RAM: 32 GB
Python: 3.11
Torch: 2.7.1 (CUDA 12.1)
Image size: 640
Model: yolov11s.pt
Batch size: 1

Limitations:
GPU memory maxes out easily
Can’t go beyond yolov11s
Slow inference (2–4 FPS on 720p video)
No shared memory benefits
I’m planning to move to an RTX 4090 (24GB) and wanted to ask:
Will I be able to run yolov11x with high-res input (1280x720 or 1920x1080)?

Can I get real-time inference (30+ FPS) on video?
Any known issues when switching from 1650 to 4090 with YOLOv11?
Appreciate any input or reference configs!

Thanks

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AbdulsamadHussain, thank you for sharing your detailed setup and interest in scaling YOLO11 inference 🚀! For in-depth performance and scalability questions, we recommend reviewing the [Docs](https://docs.ultralytics.com/), where you’ll find extensive guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, as well as hardware recommendations and optimization tips.

If you encounter any 🐛 bugs during your upgrade or have performance issues, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so our team can assist you more efficiently.

For custom training or deployment ❓ questions, sharing sample input images, configs, or inference logs will help us give you the best advice. Make sure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for optimal performance.

You’re also welcome to engage with the Ultralytics community:
- Join real-time conversations on [Discord](https://discord.com/invite/ultralytics) 🎧
- Ask or answer questions on [Discourse](https://community.ultralytics.com/)
- Share experiences on our [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

To ensure you’re leveraging the latest performance improvements, upgrade to the newest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Run YOLO in any of these up-to-date environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help you get started quickly. An Ultralytics engineer will review your query and provide further assistance soon 😊

**Reply by glenn-jocher** [reply]
@AbdulsamadHussain great questions about scaling to RTX 4090! Based on the performance benchmarks, you should see significant improvements:

**YOLO11x on RTX 4090**: Yes, you'll easily run yolov11x with 1280x720 or even 1920x1080 input with 24GB VRAM. The RTX 4090 has ~6x more VRAM and much higher compute capability than your GTX 1650.

**Real-time inference**: Absolutely achievable. YOLO11x runs at ~11.3ms per frame on a T4 GPU according to our benchmarks, and RTX 4090 is significantly faster than T4, so you should easily hit 30+ FPS even with higher resolution inputs.

**Switching considerations**: No major issues expected, but consider using TensorRT export for maximum performance on RTX 4090 - you can export with `model.export(format='engine')` and see substantial speed gains over standard PyTorch inference.

For your video pipeline, I'd also suggest experimenting with batch processing if your use case allows, as the RTX 4090's memory will support larger batch sizes for even better throughput.

**Comment by Y-T-G** [comment]
> My app performs detections and saves screenshots, outputs CSVs, and displays real-time logs.

If you're performing all of these in the same thread, then that's bad. I/O operations should run in a different thread to avoid blocking inference.

And you should switch to TensorRT FP16 inference if you haven't already.
